org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ResourceLoaderProviders.java

```diff
@@ -42,15 +44,39 @@ public Sorter getResourceSorter() {
 
 	/** Returns a loader with a parallelization degree of 2 to 4 (depending on how many processors are available). */
 	public static Provider<IResourceLoader> getParallelLoader() {
+		int nThreads = getNumberOfThreads();
+		return getParallelLoader(nThreads, 0);
+	}
+
+	/**
+	 * since 2.12
+	 */
+	public static Provider<IResourceLoader> getParallelLoader(long timeoutTime, TimeUnit timeoutUnit) {
+		int nThreads = getNumberOfThreads();
+		return getParallelLoader(nThreads, 0, timeoutTime, timeoutUnit);
+	}
+
+	/**
+	 * calculates the number of threads to be used for parallel resource loading
+	 * since 2.12
+	 */
+	public static int getNumberOfThreads() {
```

Comment:
For this unlikely case they could duplicate the method, or we introduce a provider. It is not a method that I would see as a method of ResourceLoaderProviders.